packages/core/notifications/core-notifications-browser-internal/src/toasts/telemetry/event_reporter.ts

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { type AnalyticsServiceStart } from '@kbn/core/public';
+import { EventMetric, FieldType } from './event_types';
+
+export class EventReporter {
+  private reportEvent: AnalyticsServiceStart['reportEvent'];
+
+  constructor({ analytics }: { analytics: AnalyticsServiceStart }) {
+    this.reportEvent = analytics.reportEvent;
+  }
+
+  onDismissToast({
+    recurrenceCount,
+    toastMessage,
+  }: {
+    recurrenceCount: number;
+    toastMessage: string;
+  }) {
+    this.reportEvent(EventMetric.TOAST_DISMISSED, {
+      [FieldType.RECURRENCE_COUNT]: recurrenceCount,
+      [FieldType.TOAST_MESSAGE]: toastMessage,
+    });
+  }
+
+  onClearAllToasts() {
```

Comment:
after speaking with @sixstringcode I've opted for a different approach that uses only the `onDismiss` method for all toast dismiss scenarios; when a single toast gets dismissed, when the display lifetime of a toast elapses with any interaction from a user and also when the clear all button is clicked, since we aren't particular about the differentiation on how the toast gets dismissed. what remains a constant across these scenarios is that the event only gets sent when recurrence count > 1 on the same toast ID.

Reply:
Thanks for digging into this. Awesome work, @eokoneyo!